src/main/java/org/broadinstitute/consent/http/models/User.java

```
@@ -349,4 +351,21 @@ public boolean doesUserHaveAnyRoleInSet(EnumSet<UserRoles> userRoles) {
         return getUserRoleIdsFromUser().stream().anyMatch(queriedRoleIds::contains);
     }
 
+    @Transient
+    public void checkIfUserHasRole(String roleName, Integer dacId) {
+        UserRoles role = UserRoles.getUserRoleFromName(roleName);
+        List<UserRole> roles = getRoles();
+        List<UserRole> targetRoles = roles.stream()
+                .filter((r) -> {
+                    return r.getName().equals(role.getRoleName())
+                            && r.getRoleId().equals(role.getRoleId())
+                            && Objects.equals(r.getDacId(), dacId);
+                })
+                .collect(Collectors.toList());
+        if (targetRoles.isEmpty()) {
+            // Intentionally giving 404 to avoid giving permissions insights
+            throw new NotFoundException("User role not found");
+        } 
```

I think it makes more sense to keep the model class free of this kind of exception and throw it at the resource class level instead. This way, API specific errors are not propagated through the layers of the business logic.